skippkg-finder.py

```diff
@@ -0,0 +1,333 @@
+        package.
+        """
+
+        pkglist = {}
+        packageinfo = {}
+        query = {'expand': 1}
+        root = ET.parse(http_GET(makeurl(self.apiurl, ['source', project],
+                                 query=query))).getroot()
+        for i in root.findall('entry'):
+            pkgname = i.get('name')
+            orig_project = i.get('originproject')
+            is_incidentpkg = False
+            if pkgname.startswith('000') or\
+                    pkgname.startswith('_') or\
+                    pkgname.startswith('patchinfo.') or\
+                    pkgname.startswith('skelcd-') or\
+                    pkgname.startswith('installation-images') or\
+                    pkgname.endswith('-mini'):
+                continue
+            # ugly hack for go1.x incidents as the name would be go1.x.xxx
+            if '.' in pkgname and re.match(r'[0-9]+$', pkgname.split('.')[-1]) and \
+                    orig_project.startswith('SUSE:') and orig_project.endswith(':Update'):
+                is_incidentpkg = True
+                if pkgname.startswith('go1') or pkgname.startswith('bazel0') or \
+                        pkgname.startswith('dotnet') or pkgname.startswith('ruby2'):
+                    if not (pkgname.count('.') > 1):
+                        is_incidentpkg = False
+
+            # if an incident then update the package origin info
+            if is_incidentpkg:
+                orig_name = re.sub(r'\.[0-9]+$', '', pkgname)
+                incident_number = int(pkgname.split('.')[-1])
+                if orig_name in pkglist and pkglist[orig_name]['Project'] == orig_project:
+                    if re.match(r'[0-9]+$', pkglist[orig_name]['Package'].split('.')[-1]):
+                        old_incident_number = int(pkglist[orig_name]['Package'].split('.')[-1])
+                        if incident_number > old_incident_number:
+                            pkglist[orig_name]['Package'] = pkgname
+                    else:
+                        pkglist[orig_name]['Package'] = pkgname
+            else:
+                pkglist[pkgname] = {'Project': orig_project, 'Package': pkgname}
+
+        if by_project:
+            for pkg in pkglist.keys():
+                if pkglist[pkg]['Project'].startswith('SUSE:') and self.is_sle_specific(pkg):
+                    continue
+                if pkglist[pkg]['Project'] not in packageinfo:
+                    packageinfo[pkglist[pkg]['Project']] = []
+                if pkglist[pkg]['Package'] not in packageinfo[pkglist[pkg]['Project']]:
+                    packageinfo[pkglist[pkg]['Project']].append(pkglist[pkg]['Package'])
+            return packageinfo
+
+        return pkglist
+
+    def get_project_binary_list(self, project, repository, arch, package_binaries={}):
+        """
+        Returns binarylist of a project
+        """
+
+        if project.startswith('SUSE:'):
+            repository = 'pool'
```

Comment:
why this overwrite is needed?

Reply:
That is part of CtLG design, SLE RPMs has injected to pool repository on OBS, not standard repository, so you will not find RPMs in standard repo if SLE: namespace project.